### ISSUE TITLE
only reproduce 42.94 in miniimagenet

### ISSUE THREAD

**Issue by xmengli999** [dragen1860/MAML-Pytorch#19]
Hi, Thanks for your excellent work!
I only reproduce 42.94 using the default parameters. I guess the reason maybe that the code does not contain the 2nd derivatives. The original authors of MAML using 
`if FLAGS.stop_grad:
                    grads = [tf.stop_gradient(grad) for grad in grads]`

Is this the reason for the error?

**Comment by dragen1860** [comment]
I guess probably not.
the `stop_gradient` is for using 1-order derivative approximation. I did not includes it since I only implement the 2nd order version, which should performance better.

**Comment by xmengli999** [comment]
Thanks for  your reply!
I achieved 0.4644 with more epochs. 

A few more question, I noticed `grad = torch.autograd.grad(loss, fast_weights)` can be used to calculate the 1nd order gradient and then use `fast_weights = list(map(lambda p: p[1] - self.update_lr * p[0], zip(grad, fast_weights)))` to calculate the 2nd gradient. 

if we only use `grad = torch.autograd.grad(loss, fast_weights, create_graph=True)`, is it can be used to calculate 2nd gradient?
